renderlib/RotateTool.cpp

```diff
@@ -0,0 +1,367 @@
+#include "RotateTool.h"
+
+struct ManipColors
+{
+  static constexpr glm::vec3 xAxis = { 1.0f, 0.0f, 0.0f };
+  static constexpr glm::vec3 yAxis = { 0.0f, 1.0f, 0.0f };
+  static constexpr glm::vec3 zAxis = { 0.0f, 0.0f, 1.0f };
+  static constexpr glm::vec3 bright = { 1.0f, 1.0f, 1.0f };
+};
+
+static const float s_orthogonalThreshold = cos(glm::radians(89.0f));
+
+static float
+getSignedAngle(const glm::vec3& v0, const glm::vec3& v1, const glm::vec3& vN)
+{
+  // get signed angle between the two vectors using (1,0,0) as plane normal
+  // atan2 ( (v0 X v1) dot (vN), v0 dot v1 )
+
+  float dp = dot(v0, v1);
+  float angle = atan2(dot(cross(v0, v1), vN), dp);
+  return angle;
+}
```

Comment:
I was looking up formulas for getting the angle between two 3D vectors and it looks like a lot of them do:
```
arccos( (v0 dot v1) / (|v0| * |v1|)
```
Is this calculating the angle between the vectors when projected onto the plane given by vN?

Reply:
The comment is wrong about (1,0,0); It should say `using vN as plane normal`.  I'll fix that.

The acos(dot product) is resulting in a function that doesn't know which direction the angle goes in.  acos(dot(x,y)) == acos(dot(y,x)).  We want a SIGNED angle based on sweeping a rotation FROM one vector to the other.   The cross product is order-dependent and cross(a,b) = - cross(b,a) which gives us the proper sense of rotating clockwise vs counterclockwise.   It does the projection just in case the vectors v0,v1 are not in plane.  vN also gives us the sense of which direction should be considered positive.    

cross product is proportional to sin, and dot product is proportional to cos, therefore we can do an atan with both sin and cos as parameters.